tests/lib/tools/snaps-state

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: pack-local <snap-name>"
+    echo "       install-local <snap-name> [OPTIONS]"
+    echo "       install-local-as <snap-name> <dest-name> [OPTIONS]"
+    echo "       is-classic-confinement-supported"
+    echo ""
+    echo "Available options:"
+    echo "  --devmode --jailmode --classic"
+    echo ""
+    echo "Pack and install commands save the packed snap for future uses,"
+    echo "which is reused on the following calls."
+    echo "The paths for locating the sources of the snaps to either pack or"
+    echo "install are the local path and then 'tests/lib/snaps/'"
+}
+
+pack_local() {
+    local SNAP_NAME="$1"
+    local SNAP_DIR="$TESTSLIB/snaps/${SNAP_NAME}"
+    if [ $# -gt 1 ]; then
+        SNAP_DIR="$2"
+    fi
+    local SNAP_FILE="${SNAP_DIR}/${SNAP_NAME}_1.0_all.snap"
```

Comment:
Not a fan of the hard-coded version string. I personally use `1` for the version, not `1.0`. Not sure what to do about this yet.

Reply:
The only problem is that we have many tests (about 40) already using this 1.0_all.snap

So the change will be much bigger if we change that name. What do you think @zyga ?

Reply:
I think it would be best not to change the tests but make this part smarter to know what the version is.

Reply:
sure, I updated that